--- CMakeLists.txt.original	2015-12-03 23:35:45.727569487 +0100
+++ CMakeLists.txt	2015-12-03 23:37:36.214564866 +0100
@@ -648,6 +648,17 @@
 # Enable packaging
 ####################################################################
 include (CPack)
+INSTALL(FILES CMakeLists.txt AxelConfig.cmake
+       DESTINATION "."
+       )
+
+set (AXEL_LIBRARY_DIR ${AXEL_LIBRARY_DIR} /opt/axel-sdk/lib/ )
+
+INSTALL(FILES "${PROJECT_BINARY_DIR}/axlCoreExport.h"
+       DESTINATION "include/axlCore/"
+       )
+
+INSTALL(DIRECTORY "${PROJECT_BINARY_DIR}/plugins" DESTINATION /opt/axel-sdk/ )
 
 ####################################################################
 # Installer
