--- sci-mathematics/axel-sdk-2015.07/work/axel-sdk-2015.07/axel/CMakeLists.txt~	2015-07-22 18:04:08.000000000 +0200
+++ sci-mathematics/axel-sdk-2015.07/work/axel-sdk-2015.07/axel/CMakeLists.txt	2015-07-22 18:05:05.371773529 +0200
@@ -491,3 +491,8 @@
 if(ENABLE_TEST)
    add_subdirectory(tests)
 endif(ENABLE_TEST)
+
+install(DIRECTORY src/axlCore
+        DESTINATION include
+        FILES_MATCHING PATTERN "*.h" 
+)
\ No newline at end of file
